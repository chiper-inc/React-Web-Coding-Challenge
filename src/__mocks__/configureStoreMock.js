import { compose, applyMiddleware, createStore } from 'redux'
import { INITIAL_STATE, reducer } from '../store/reducer'
import thunk from 'redux-thunk'

const middleware = [thunk]

export const configureStoreMock = () => createStore(
  reducer,
  INITIAL_STATE,
  compose(applyMiddleware(...middleware))
)
