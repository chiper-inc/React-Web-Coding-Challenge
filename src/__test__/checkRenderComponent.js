import { mount } from 'enzyme'
import { create } from 'react-test-renderer'

export const checkRenderComponent = (expect, component) => {
  const renderedComp = mount(component)
  expect(renderedComp.length).toEqual(1)
}

export const checkSnapShot = (expect, component) => {
  const renderComp = create(component)
  expect(renderComp.toJSON()).toMatchSnapshot()
}
