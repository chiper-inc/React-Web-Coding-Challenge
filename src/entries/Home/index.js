import React, { useEffect } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { StolenBikesContainer } from "../../components/StolenBikesContainer";
import { requestStolenBikes, requestPage } from "../../store/actionCreators";
import { PAGE_LEN_API, PAGE_LEN_UI } from "./configPagination";

const getTotal = (bikes) =>  bikes.length >= PAGE_LEN_API ? `+${PAGE_LEN_API}`: bikes.length;

const StolenBikesContainerC = ({
  loading,
  error,
  stolenBikes,
  page,
  actions,
}) => {
  useEffect(() => {
    actions.requestStolenBikes(1,PAGE_LEN_API);
  }, [actions]);

  const handleChangePage = (page) => {
    console.log(page)
    actions.requestPage(page);
  }
  return (
    <StolenBikesContainer
      bikes={stolenBikes}
      page={page}
      lengthPage={PAGE_LEN_UI}
      total={getTotal(stolenBikes)}
      onChangePage={handleChangePage}
      loading={loading}
      error={error}
    />
  );
};

const mapStateToProps = (state) => {
  const { loading, error, stolenBikes, page, lengthPage } = state;

  return {
    loading,
    error,
    page,
    lengthPage,
    stolenBikes,
  };
};

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators({ requestStolenBikes, requestPage }, dispatch),
});

export const Home = connect(
  mapStateToProps,
  mapDispatchToProps
)(StolenBikesContainerC);
