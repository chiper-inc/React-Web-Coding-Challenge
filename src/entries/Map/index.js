import React from 'react'

export const Map = () => <div>Hello World!</div>
