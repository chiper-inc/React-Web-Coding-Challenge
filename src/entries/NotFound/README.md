from https://codepen.io/kdbkapsere/pen/oNXLbqQ