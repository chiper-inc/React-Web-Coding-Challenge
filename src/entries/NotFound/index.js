import React from 'react'
import { NavLink } from 'react-router-dom'
import { HOME } from '../../routes'
import { PageInfo } from '../../components/PageInfo'

export const NotFound = () => (
  <PageInfo>
    <h1>404</h1>
    <h2>UH OH! You're lost.</h2>
    <p>
      The page you are looking for does not exist. How you got here is a
      mystery. But you can click the button below to go back to the homepage.
    </p>
    <NavLink
      to={HOME}
      isActive={() => false}
      className='btn btn-outline-secondary'
    >
      HOME
    </NavLink>
  </PageInfo>
)
