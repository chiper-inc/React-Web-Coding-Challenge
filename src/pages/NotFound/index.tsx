const NotFound = () => (
  <section><p>Sitio no encontrado</p></section>
)

export default NotFound;
