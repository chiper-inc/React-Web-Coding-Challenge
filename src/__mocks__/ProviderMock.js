import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { configureStoreMock } from './configureStoreMock'

const store = configureStoreMock()

export const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <BrowserRouter>{children}</BrowserRouter>
  </Provider>
)
