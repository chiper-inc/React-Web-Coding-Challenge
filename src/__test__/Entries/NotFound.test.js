/* eslint-disable no-undef */
import React from 'react'
import { NotFound } from '../../entries/NotFound'
import { checkRenderComponent, checkSnapShot } from '../checkRenderComponent'
import { ProviderMock } from '../../__mocks__/ProviderMock'

describe('<NotFound />', () => {
  test('Render of <NotFound />', () => {
    checkRenderComponent(
      expect,
      <ProviderMock>
        <NotFound />
      </ProviderMock>
    )
  })
})

describe('<NotFound/> Snapshot', () => {
  test('Check UI of NotFound', () => {
    checkSnapShot(
      expect,
      <ProviderMock>
        <NotFound />
      </ProviderMock>
    )
  })
})
