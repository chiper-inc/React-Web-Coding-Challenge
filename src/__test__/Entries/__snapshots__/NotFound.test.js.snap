// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> Snapshot Check UI of NotFound 1`] = `
<div
  className="sc-AxjAm kSkoow container row"
>
  <div
    className="col-md-6 align-self-center"
  >
    <img
      src="austronat.svg"
    />
  </div>
  <div
    className="col-md-6 align-self-center"
  >
    <h1>
      404
    </h1>
    <h2>
      UH OH! You're lost.
    </h2>
    <p>
      The page you are looking for does not exist. How you got here is a mystery. But you can click the button below to go back to the homepage.
    </p>
    <a
      aria-current={null}
      className="btn btn-outline-secondary"
      href="/"
      onClick={[Function]}
    >
      HOME
    </a>
  </div>
</div>
`;
